src/rqt_service_caller/service_caller_widget.py

```diff
@@ -80,9 +80,16 @@ def __init__(self):
         self.request_tree_widget.itemChanged.connect(self.request_tree_widget_itemChanged)
 
     def save_settings(self, plugin_settings, instance_settings):
+        instance_settings.set_value('current_service_name', self._service_info['service_name'])
```

Comment:
`self._service_info` is initialized with `None`. Doesn't this fail if nothing was selected before the perspective is being saved?

Reply:
`on_refresh_services_button_clicked` is called directly after this initialization. This means all available services (at least the _/rosout/..._ services) are added to the ComboBox. The index of the ComboBox will change immedeatly and `self._service_info` gets valid. So there will be always a service selected and saving the perspective will not fail.
I could add an additional null check (if you wish) but I think it is not required.